packages/in_app_purchase/example/lib/main.dart

```diff
@@ -362,20 +362,17 @@ class _MyAppState extends State<MyApp> {
 
   void _listenToPurchaseUpdated(List<PurchaseDetails> purchaseDetailsList) {
     purchaseDetailsList.forEach((PurchaseDetails purchaseDetails) async {
-      await InAppPurchaseConnection.instance.consumePurchase(purchaseDetails);
       if (purchaseDetails.status == PurchaseStatus.pending) {
         showPendingUI();
       } else {
         if (purchaseDetails.status == PurchaseStatus.error) {
           handleError(purchaseDetails.error);
-          return;
         } else if (purchaseDetails.status == PurchaseStatus.purchased) {
           bool valid = await _verifyPurchase(purchaseDetails);
           if (valid) {
             deliverProduct(purchaseDetails);
           } else {
             _handleInvalidPurchase(purchaseDetails);
-            return;
```

Comment:
I am actually not 100% sure about this return removal. I am guessing probably it is better to throw an exception in _handleInvalidPurchase() as client, but if something is invalid and we try to complete or consume the purchase that part would fail, is my assumption correct on that?

Reply:
I think we can leave the return as it is. If something is invalid, we should escape the code as soon as possible. It is also more readable in my opinion.

Reply:
Yea I agree as well. I will fix it fast.